### PR TITLE
WebView controller let's you clear cache

### PR DESCRIPTION
**Pull request by iskakaushik**

### CODE REVIEW

**Comment by iskakaushik** [comment]
@amirh I was trying to come up with ways to effectively test this or have an example for this. I wasn't able to find any JS methods that let you list the entries of the browser cache. Let me know if you have any ideas..

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 1]
```diff
@@ -394,6 +394,16 @@ class WebViewController {
     return _channel.invokeMethod("reload");
   }
 
+  /// Clears the resource cache.
+  ///
+  /// Note that the cache is per-application, so this will clear the cache for all WebViews used.
+  Future<void> clearCache() async {
```
Comment (amirh) [inline packages/webview_flutter/lib/webview_flutter.dart]: Looks like Android and iOS have some fine grained control over what exactly is being cleared.
I think it's ok to clear all (if we want to provide finer grained control we could add optional parameters here without it being a breaking change), but lets note in the documentation exactly what is being cleared on each platform.

Comment (iskakaushik) [reply packages/webview_flutter/lib/webview_flutter.dart]: @amirh noted in the docs.

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 2]
```diff
@@ -394,6 +394,24 @@ class WebViewController {
     return _channel.invokeMethod("reload");
   }
 
+  /// Clears all of the following caches.
```
Comment (amirh) [inline packages/webview_flutter/lib/webview_flutter.dart]: This shows as the summary line in the Dartdoc site, it should be self contained, probably should just say "Clears all caches used by the [WebView]."

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 3]
```diff
@@ -394,6 +394,24 @@ class WebViewController {
...
+  ///
+  /// There are multiple caches we need to be aware of.
+  ///	1. Browser HTTP Cache.
+  ///	2. Caches API caches. https://developers.google.com/web/fundamentals/instant-and-offline/web-storage/cache-api,
```
Comment (amirh) [inline packages/webview_flutter/lib/webview_flutter.dart]: Caches API caches? typo?

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 4]
```diff
@@ -394,6 +394,24 @@ class WebViewController {
...
+  ///    These are not yet supported in IOS WkWebView. Service workers tend to use this cache.
```
Comment (amirh) [inline packages/webview_flutter/lib/webview_flutter.dart]: nit: iOS

File: packages/webview_flutter/lib/webview_flutter.dart (hunk 3 above)
Comment (amirh) [reply packages/webview_flutter/lib/webview_flutter.dart]: Maybe would be more clear as:
```dart
/// [Caches API](https://developers.google.com/web/fundamentals/instant-and-offline/web-storage/cache-api) cache.
```

File: packages/webview_flutter/lib/webview_flutter.dart [hunk 5]
```diff
@@ -394,6 +394,24 @@ class WebViewController {
     return _channel.invokeMethod("reload");
   }
 
+  /// Clears all of the following caches.
+  ///
+  /// There are multiple caches we need to be aware of.
```
Comment (amirh) [inline packages/webview_flutter/lib/webview_flutter.dart]: Suggestion: "The following caches are cleared:"

**Pull request merged**